controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -310,8 +310,54 @@ func (r *StorageClassClaimReconciler) reconcileConsumerPhases() (reconcile.Resul
 		// Update the StorageClassClaim status.
 		r.storageClassClaim.Status.Phase = v1alpha1.StorageClassClaimReady
 
+	} else {
+
+		// Reconciliation needed if the StorageClassClaim is present.
+		if err = r.get(r.storageClassClaim); !errors.IsNotFound(err) {
+
+			// Deletion phase.
+			// Update the StorageClassClaim status.
+			r.storageClassClaim.Status.Phase = v1alpha1.StorageClassClaimDeleting
+
+			// Delete StorageClass.
+			existingStorageClass := &storagev1.StorageClass{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: r.storageClassClaim.Name,
+				},
+			}
+			if err = r.get(existingStorageClass); err == nil {
+
+				// Remove finalizer from StorageClass.
+				existingStorageClass.SetFinalizers(remove(existingStorageClass.Finalizers, StorageClassFinalizer))
+				if err := r.update(existingStorageClass); err != nil {
+					return reconcile.Result{}, fmt.Errorf("failed to remove finalizer from StorageClass: %s", err)
+				}
+				if err = r.delete(existingStorageClass); err != nil {
+					return reconcile.Result{}, fmt.Errorf("failed to delete StorageClass %s: %s", existingStorageClass.Name, err)
+				}
+			} else if errors.IsNotFound(err) {
+				r.log.Info("StorageClass already deleted.")
+			} else {
+				return reconcile.Result{}, fmt.Errorf("failed to get StorageClass %s: %s", existingStorageClass.Name, err)
+			}
+
+			// Call `RevokeStorageClassClaim` service on the provider server with StorageClassClaim as a request message.
+			// Check if StorageClassClaim is still exists (it might have been manually removed during the StorageClass
+			// removal above).
+			if !errors.IsNotFound(r.get(r.storageClassClaim)) {
+				_, err = providerClient.RevokeStorageClassClaim(r.ctx, r.storageCluster.Status.ExternalStorage.ConsumerID, r.storageClassClaim.Name)
```

Comment:
What will happen if there are still PVs using the resources provisioned by this storageclassclaim? I think we first need to validate customer has deleted all PVs before sending the remove command.

Reply:
Removed the PV check based on this: https://github.com/red-hat-storage/ocs-operator/pull/1608#discussion_r860739646, PTAL.

Reply:
There is a difference between recreation of a storageclass (which is a superficial change) and revoking a claim which is going to trigger the deletion of backend resources. In the case of the latter I think we do need to validate usage before revoking